### PR TITLE
G2 2023 v3 issue#13396

### PR DESCRIPTION
**Pull request by a21ponkr**
I have written tests for services in courseedservice #13396

### CODE REVIEW

File: backend-models/service-tests/courseedservice-test.txt [hunk 1]
```diff
@@ -0,0 +1,1284 @@
...
+* pre-req:
+********* 
+
+if(checklogin()){
+	$isSuperUserVar=isSuperUser($userid);
+
+	$ha = $isSuperUserVar;
+
+	if($ha){
+
```
Comment (a21jeaha) [inline backend-models/service-tests/courseedservice-test.txt:21]: row 16 , 18, and 20:
You have just copied the code over to the text document. This does not give any information on what state the code needs to be in for the test to be able to executed.

What actually is a pre-req is: if(checklogin()): has to be true 
                                               $ha = value of super user

What actual values does $isSuperUserVar need to have for the test to proceed. it needs to be a super user. what value is associated with a super user?

File: backend-models/service-tests/courseedservice-test.txt [hunk 2]
```diff
@@ -0,0 +1,1284 @@
...
+$query = $pdo->prepare("INSERT INTO course (coursecode,coursename,visibility,creator, hp, courseGitURL) VALUES(:coursecode,:coursename,0,:usrid, 7.5, :courseGitURL)");
```
Comment (a21jeaha) [inline backend-models/service-tests/courseedservice-test.txt:22]: The query values is to be supplied as actual input-values in an array.

Example:
///////
/ pre-req variables
//////

$pre-Values = { null,
                1885, 
                0,
                1,
                "AutomaticTest",
                "Quiz",
                "2023-04-27 00:00:00",
                "2023-04-29 00:00:00",
                2,
                "1337",
                "2023-04-17",
                "{&quot;deadline1&quot;:&quot;2023-04-27 0:0&quot;,&quot;comment1&quot;:&quot;&quot;,&quot;deadline2&quot;:&quot;&quot;,&quot;comment2&quot;:&quot;&quot;,&quot;deadline3&quot;:&quot;&quot;,&quot;comment3&quot;:&quot;&quot;}", 
                0 }; 


SQL-query:  INSERT INTO quiz(cid,autograde,gradesystem,qname,quizFile,qrelease,deadline,creator,vers,qstart,jsondeadline,`group`) 
            VALUES ($pre-Values[1], $pre-Values[2], $pre-Values[3], $pre-Values[4], $pre-Values[5], $pre-Values[6], $pre-Values[7], $pre-Values[8], $pre-Values[9], $pre-Values[10], $pre-Values[11] );
            
the used values need to be correct, you should be able to see the added course in LenaSYS

File: backend-models/service-tests/courseedservice-test.txt [hunk 3]
```diff
@@ -0,0 +1,1284 @@
...
+
+*******
+* bind parameters
+******
+
+$query->bindParam(':usrid', $userid);
+$query->bindParam(':coursecode', $coursecode);
+$query->bindParam(':coursename', $coursename);
+$query->bindParam(':courseGitURL', $courseGitURL);
+
```
Comment (a21jeaha) [inline backend-models/service-tests/courseedservice-test.txt:32]: no bind parameters!

this text file is supposed to be language indapendant

File: backend-models/service-tests/courseedservice-test.txt [hunk 4]
```diff
@@ -0,0 +1,1284 @@
...
+*******
+* show error if we can't update entries
```
Comment (a21jeaha) [inline backend-models/service-tests/courseedservice-test.txt:34]: is this ever used?

**Review by a21jeaha — CHANGES_REQUESTED** [review]
i have given some comments regarding changes. i have not gone trough the entire file.
but you need to restructure the file to follow the standard set in this file
/backend-models/service-tests/duggaedservice-tests.txt
make the changes then ill read trough the rest.

**Comment by a21ponkr** [comment]
I do not know how to do this issue and i think it's better for someone more knowledgeable to do it rather than it taking too long to be finished.

**Comment by d21erisu** [comment]
since a21ponkr threw in the towel and a21oscgu gives it a try, i closed this PR. 
Issue: https://github.com/HGustavs/LenaSYS/issues/13396

**Pull request closed without merging**